utils/utils.go

```diff
@@ -243,6 +243,60 @@ func CustomizeService(svc *corev1.Service, ba common.BaseComponent) {
 	}
 }
 
+func CustomizeProbes(container *corev1.Container, ba common.BaseComponent) {
+	probesConfig := ba.GetProbes()
+
+	// Probes not defined -- reset all probesConfig to nil
+	if probesConfig == nil {
+		container.ReadinessProbe = nil
+		container.LivenessProbe = nil
+		container.StartupProbe = nil
+		return
+	}
+
+	customizeProbe(&container.ReadinessProbe, probesConfig.GetReadinessProbe(), probesConfig.GetDefaultReadinessProbe, ba)
```

Comment:
The only way I think this can panic is if container is nil. If it is not nil, then the address of the probe should be valid. However, I did refactor the code to make it more readable though.